cmake/BuildLibSanitizers.cmake

```diff
@@ -23,36 +23,65 @@ macro(GEN_BUILD_TARGET name libsanitizer_path libfuzzer_path
 
   # Strip preinit object files in static libraries, otherwise a message
   # `.preinit_array section is not allowed in DSO` will prevent building DSO.
-  add_custom_target(strip_lib_${name}
-    COMMENT "Strip sanitizer library ${name}"
-    COMMAND ${CMAKE_C_COMPILER_AR} d ${libsanitizer_name} ${strip}
-    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-    BYPRODUCTS ${libsanitizer_name}
-    DEPENDS copy_libs_${name}
-  )
+  if(strip AND NOT "${strip}" STREQUAL "")
+    add_custom_target(strip_lib_${name}
+      COMMENT "Strip sanitizer library ${name}"
+      COMMAND ${CMAKE_C_COMPILER_AR} d ${libsanitizer_name} ${strip}
+      WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+      BYPRODUCTS ${libsanitizer_name}
+      DEPENDS copy_libs_${name}
+    )
+  else()
+    add_custom_target(strip_lib_${name}
+      COMMENT "Skip stripping sanitizer library ${name} (no objects to strip)"
+      DEPENDS copy_libs_${name}
+    )
+  endif()
```

Comment:
I would leave only variable with command under condition and call `add_custom_target()` only once.

Reply:
Fixed